includes/embeds/class-amp-playlist-embed-handler.php

```diff
@@ -0,0 +1,328 @@
+		if ( ! isset( $data['tracks'] ) ) {
+			return '';
+		}
+		$container_id   = 'ampPlaylistCarousel' . self::$playlist_id;
+		$selected_slide = $container_id . '.selectedSlide';
+		self::$playlist_id++;
+
+		$this->enqueue_styles();
+		ob_start();
+		?>
+		<div class="wp-playlist wp-audio-playlist wp-playlist-light">
+			<amp-carousel id="<?php echo esc_attr( $container_id ); ?>" [slide]="<?php echo esc_attr( $selected_slide ); ?>" height="<?php echo esc_attr( self::CAROUSEL_HEIGHT ); ?>" width="auto" type="slides">
+				<?php
+				foreach ( $data['tracks'] as $track ) :
+					$title      = $this->get_title( $track );
+					$image_url  = isset( $track['thumb']['src'] ) ? $track['thumb']['src'] : '';
+					$dimensions = $this->get_thumb_dimensions( $track );
+					?>
+					<div>
+						<div class="wp-playlist-current-item">
+							<?php if ( $image_url ) : ?>
+								<amp-img src="<?php echo esc_url( $image_url ); ?>" height="<?php echo esc_attr( $dimensions['height'] ); ?>" width="<?php echo esc_attr( $dimensions['width'] ); ?>"></amp-img>
+							<?php endif; ?>
+							<div class="wp-playlist-caption">
+								<span class="wp-playlist-item-meta wp-playlist-item-title"><?php echo esc_html( $title ); ?></span>
+							</div>
+						</div>
+						<amp-audio width="auto" height="50" src="<?php echo esc_url( $track['src'] ); ?>"></amp-audio>
+					</div>
+				<?php endforeach; ?>
+			</amp-carousel>
+			<?php $this->tracks( 'audio', $container_id, $data['tracks'] ); ?>
+		</div>
+		<?php
+		return ob_get_clean();
+	}
+
+	/**
+	 * Gets an AMP-compliant video playlist.
+	 *
+	 * This uses similar markup to the native playlist shortcode output.
+	 * So the styles from wp-mediaelement.min.css will apply to it.
+	 *
+	 * @global int $content_width
+	 * @param array $data Data.
+	 * @return string $video_playlist Markup for the video playlist.
+	 */
+	public function video_playlist( $data ) {
+		global $content_width;
+		if ( ! isset( $data['tracks'], $data['tracks'][0]['src'] ) ) {
+			return '';
+		}
+		$playlist = 'playlist' . self::$playlist_id;
+		self::$playlist_id++;
+
+		$amp_state = array(
+			'currentVideo' => '0',
+		);
+		foreach ( $data['tracks'] as $index => $track ) {
+			$amp_state[ $index ] = array(
+				'videoUrl' => $track['src'],
+				'thumb'    => isset( $track['thumb']['src'] ) ? $track['thumb']['src'] : '',
+			);
+		}
+
+		$dimensions = isset( $data['tracks'][0]['dimensions']['resized'] ) ? $data['tracks'][0]['dimensions']['resized'] : null;
+		$width      = isset( $dimensions['width'] ) ? $dimensions['width'] : $content_width;
+		$height     = isset( $dimensions['height'] ) ? $dimensions['height'] : null;
+
+		$this->enqueue_styles();
+		ob_start();
+		?>
+		<div class="wp-playlist wp-video-playlist wp-playlist-light">
+			<amp-state id="<?php echo esc_attr( $playlist ); ?>">
+				<script type="application/json"><?php echo wp_unslash( wp_json_encode( $amp_state ) ); // WPCS: XSS ok. ?></script>
```

Comment:
Note that `wp_unslash()` isn't right here. If you want to not have slashes in the output, you should use `JSON_UNESCAPED_SLASHES`. Otherwise, the slashes are important for security since it ensures that if a string contains `</script>` it will get output as `<\/script>` and prevented from breaking out of the scriptl

I'm amending the PR.